go-controller/pkg/cni/cnismartnic.go

```diff
@@ -68,10 +68,26 @@ func (bp *SmartNicPlugin) CmdAdd(args *skel.CmdArgs) error {
 	}
 	kube := &kube.Kube{KClient: clientset}
 
-	// 4. Set smart-nic pod annotation of PF index and VF index
-	smartNicAnnotation := fmt.Sprintf("pf=%s;vf=%d", pfPciAddress, vfindex)
-	err = kube.SetAnnotationsOnPod(podReq.PodNamespace, podReq.PodName, map[string]string{
-		SmartNicConnectionDetails: smartNicAnnotation, "sandbox": podReq.SandboxID})
+	// 4. Set smart-nic connection-details pod annotation
+	var domain, bus, dev, fn int
+	if parsed, err := fmt.Sscanf(
+		pfPciAddress, "%04x:%02x:%02x.%d", &domain, &bus, &dev, &fn); err != nil || parsed != 4 {
+		return fmt.Errorf("failed to parse PF PCI address %s. Unexpected format", pfPciAddress)
+	}
```

Comment:
nit, I think that this is easier to read
```
        parsed, err := fmt.Sscanf(pfPciAddress, "%04x:%02x:%02x.%d", &domain, &bus, &dev, &fn)
        if err != nil  {
		return fmt.Errorf("error trying to parse PF PCI address %s: %v", pfPciAddress, err)

	}
        if parsed != 4 {
		return fmt.Errorf("failed to parse PF PCI address %s. Unexpected format", pfPciAddress)
	}
```

Reply:
will modify thanks!